arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# '-mcmse' enables the generation of code for the Secure state of the ARMv8-M
+# Security Extensions. This option is required when building a Secure firmware.
+zephyr_compile_options(-mcmse)
+
+if(${CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS})
+# '--section-start,.gnu.sgstubs' enables entry veneers (for Secure Entry
+# functions) to be placed inside the special output section .gnu.sgstubs.
+	zephyr_ld_options(
+		${LINKERFLAGPREFIX},--section-start,.gnu.sgstubs=${CONFIG_ARM_NSC_REGION_BASE_ADDRESS}
+	)
+
+# --out-implib and --cmse-implib instruct the linker to produce
+# an import library that consists of a relocatable file containing
+# only a symbol table with the entry veneers. The library may be used
+# when building a Non-Secure image which shall have access to Secure
+# Entry functions.
+	zephyr_ld_options(
+		${LINKERFLAGPREFIX},--out-implib=${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+	)
+
+	zephyr_ld_options(
+		${LINKERFLAGPREFIX},--cmse-implib
+	)
```

Comment:
Could you double-check that the indentation here is following the (de-facto) standard.

Reply:
Will fix this, thanks.